File: solutions/javascript/src/algorithm/stack/Exercise2.js
import DynamicStackADT from "../../structure/dynamic/DynamicStackADT.js";

function multiplyStackElements(stack, factor) {
  const aux = new DynamicStackADT();
  while (!stack.isEmpty()) {
        aux.add(stack.getElement() * factor);
        stack.remove();
  }

  // Restaurar la pila original con los valores modificados
  while (!aux.isEmpty()) {
        stack.add(aux.getElement());
        aux.remove();
  }
}

// Ejemplo de uso
const stack = new DynamicStackADT();
stack.add(1);
stack.add(2);
stack.add(3);

multiplyStackElements(stack, 10);

// Mostrar pila
while (!stack.isEmpty()) {
    console.log(stack.getElement()); // 30, 20, 10
    stack.remove();
}

File: solutions/python/algorithm/stack/Exercise2.py
from structure.dynamic.DynamicStackADT import DynamicStackADT

def multiply_stack_elements(stack, factor):
    aux = DynamicStackADT()
    while not stack.is_empty():
        aux.add(stack.getElement() * factor)
        stack.remove()

    # Restaurar pila
    while not aux.is_empty():
        stack.add(aux.getElement())
        aux.remove()

# Ejemplo de uso
stack = DynamicStackADT()
stack.add(1)
stack.add(2)
stack.add(3)

multiply_stack_elements(stack, 10)

while not stack.is_empty():
    print(stack.getElement())  # 30, 20, 10
    stack.remove()